app/models/metadata_field.rb

```diff
@@ -3,10 +3,24 @@ class MetadataField < ApplicationRecord
   has_and_belongs_to_many :projects
   has_many :metadata, dependent: :destroy
 
+  validate :metadata_field_subsets
+
   STRING_TYPE = 0
   NUMBER_TYPE = 1
   DATE_TYPE = 2
   LOCATION_TYPE = 3
+  validates :base_type, presence: true, inclusion: { in: [
+    STRING_TYPE,
+    NUMBER_TYPE,
+    DATE_TYPE,
+    LOCATION_TYPE,
+  ], }
+
+  # NOTE: not sure why these columns were not created as booleans
+  validates :force_options, inclusion: { in: [0, 1] }
+  validates :is_core, inclusion: { in: [0, 1] }
+  validates :is_default, inclusion: { in: [0, 1] }
+  validates :is_required, inclusion: { in: [0, 1] }
```

Comment:
Thanks for adding all of these additional validations!

Regarding the "booleans", I recall the reason why I didn't make them booleans was because I observed that similar previous columns were also not booleans and I just followed what was done previously (I was worried that there would be additional things to figure out when using booleans for the first time). I would definitely support a convention to use boolean columns in the future.

Reply:
Ok good to know. The advantage from what I read is that rails will convert mysql tinyint into `true` and `false`.